tests/30rooms/10redactions.pl

```diff
@@ -215,3 +223,42 @@ sub matrix_redact_event_synced
          Future->done;
       });
    };
+
+test "PUT /rooms/:room_id/redact/:event_id/:txn_id is idempotent",
+   requires => [ local_user_fixtures( 2 ),
+                 qw( can_send_message )],
+
+   do => sub {
+      my ( $creator, $sender ) = @_;
+      my $txn_id = random_transaction_id();
+      my ( $event_id_1, $event_id_2 );
+
+      make_room_and_message( [ $creator, $sender ], $sender )
+      ->then( sub {
+         my ( $room_id, $to_redact ) = @_;
+
+         $to_redact = uri_escape( $to_redact );
+
+         do_request_json_for( $creator,
+            method => "PUT",
+            uri    => "/v3/rooms/$room_id/redact/$to_redact/$txn_id",
+            content => {},
+         )
+         ->then( sub {
```

Comment:
```suggestion
         );
      })->then( sub {
```

(and corresponding unnesting below)

Reply:
I should have said that just after pushing my second set of changes, but I think that's not possible because we need `$room_id` and `$to_redact` (which are declared on line 238) in the scope on these 2 last blocks. Is there any way of structure it in a better way so these blocks can be unnested?

Reply:
oh right. you can declare the variables at the function level with

```
my $room_id;
my $to_redact;
```

... and then remove the `my` at line 238. But it may not be worth it (and it looks like other tests in this file follow the same pattern as you).

Reply:
I decided to apply your suggestion here: https://github.com/matrix-org/sytest/pull/1235/commits/3536eae46a82659eab3ffe4b2738889768f56836

I've also formatted the `then`s as you suggested on the other comment. 

Sorry for this back and forth about the formatting. Hope it's fine now!